### PR TITLE
meson: Add 0.54.0

### PR DESCRIPTION
**Pull request by michaelkuhn**
This change also improves the rpath patch we are using. Instead of never removing the rpath, we now only keep it if running within Spack to guarantee consistent behavior of Meson.

### CODE REVIEW

File: var/spack/repos/builtin/packages/meson/package.py [hunk 1]
```diff
@@ -30,4 +31,5 @@ class Meson(PythonPackage):
     # By default, Meson strips the rpath on installation. This patch disables
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/meson/package.py:31]: Can you add a description to the variant?

Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/meson/package.py:31]: To be honest, I am not quite sure what it is for. While `MesonPackage` adds dependencies on both `meson` and `ninja`, hybrid packages like `atk` only have a meson dependency and trying to build them with `meson~ninjabuild` actually fails. Should I just remove it and make `ninja` a hard dependency?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/meson/package.py:31]: Yeah, if ninja is the only backend we can support, I would remove that variant.

Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/meson/package.py:29]: I have pushed a new commit that removes the variant.

**Pull request merged**